lib/hocon/impl/abstract_config_value.rb

```diff
@@ -60,6 +157,78 @@ def require_not_ignoring_fallbacks
     end
   end
 
+  def construct_delayed_merge(origin, stack)
+    # TODO: this might not work because ConfigDelayedMerge inherits
```

Comment:
I don't know what TODO though.  :)

My plan was to leave these here until we have a test that causes them to fail, and then deal with it at that point, because if it doesn't work as-is it's probably going to be a very involved refactor that takes us pretty far away from the upstream class heirarchy, due to Ruby's limitations around circular references between classes compared to Java's.